BlueDB/src/main/java/org/bluedb/api/BlueCollection.java

```diff
@@ -9,25 +9,88 @@
 import org.bluedb.api.keys.BlueKey;
 import org.bluedb.api.keys.ValueKey;
 
+/**
+ * A BlueCollection is a persisted map of keys (of type BlueKey) to values of object type T.
+ * 
+ * A collection has a name (to distinguish between collections in a BlueDb instance), a key type, and a value type.
+ * @param <T> object type of values to be serialized into collection
+ */
 public interface BlueCollection<T extends Serializable> {
 
+	/**
+	 * Creates (or returns existing) BlueIndex that maps objects of type keyType to values in the collection.

```

Comment:
maps objects of type ValueKey (UUID, String, Integer, Long) to values in the collection

Reply:
I extends ValueKey but that's not super obvious in the javadoc. Your options for what you can use in an index isn't super obvious. I'd just like it to be more obvious. (Same with `getIndex` method)